### PR TITLE
chore(rum-core): tests for sending span destination context to server

### LINKED ISSUE #490: Add destination.service to spans for service maps feature
See https://github.com/elastic/apm/issues/180 for more
Separate but related: https://github.com/elastic/apm-agent-rum-js/issues/513

task:
- [ ] Add actual payload tests for `context.destination`
- [ ] make sure the snapshot is update with the destination specs from the APM server

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ fix elastic/apm-agent-rum-js#490
+ Tested manually with the latest APM-server master.

### CODE REVIEW

File: packages/rum-core/test/performance-monitoring/performance-monitoring.spec.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ import Span from '../../src/performance-monitoring/span'
 import { getGlobalConfig } from '../../../../dev-utils/test-config'
 import { getDtHeaderValue } from '../../src/common/utils'
 import { globalState } from '../../src/common/patching/patch-utils'
+import { patchEventHandler as originalPathHandler } from '../../src/common/patching'
```
Comment (jahtalab) [inline packages/rum-core/test/performance-monitoring/performance-monitoring.spec.js]: nit: typo `originalPatchHandler`

File: packages/rum-core/test/performance-monitoring/performance-monitoring.spec.js [hunk 2]
```diff
@@ -763,4 +764,48 @@ describe('PerformanceMonitoring', function() {
     performanceMonitoring.processAPICalls('invoke', task3)
     expect(tr.ended).toBe(true)
   })
+
+  it('should send span context destination details to apm-server', done => {
+    const transactionService = serviceFactory.getService('TransactionService')
+
+    const tr = transactionService.startTransaction('with-context', 'custom', {})
```
Comment (jahtalab) [inline packages/rum-core/test/performance-monitoring/performance-monitoring.spec.js]: The empty object is not necessary for the options.

File: packages/rum-core/test/performance-monitoring/performance-monitoring.spec.js [hunk 3]
```diff
@@ -763,4 +764,48 @@ describe('PerformanceMonitoring', function() {
...
+
+    configService.events.observe(TRANSACTION_END, () => {
+      expect(tr.spans.length).toBe(2)
+      const [span1, span2] = tr.spans
+      expect(span1.context.destination).toBeDefined()
+      expect(span2.context.destination).toBeDefined()
+
+      const payload = performanceMonitoring.convertTransactionsToServerModel([
+        tr
+      ])
```
Comment (jahtalab) [inline packages/rum-core/test/performance-monitoring/performance-monitoring.spec.js]: Maybe add checks here that the destination is added to the payload!

Comment (vigneshshanmugam) [reply packages/rum-core/test/performance-monitoring/performance-monitoring.spec.js]: There is already a check here right? https://github.com/elastic/apm-agent-rum-js/pull/539/files#diff-4f75686bfc54bd212244f355ce6f93f5R793-R794

**Comment by vigneshshanmugam** [comment]
@jahtalab You are right, Checked the server PR and added the truncation logic as well.

File: packages/rum-core/test/performance-monitoring/performance-monitoring.spec.js [hunk 4]
```diff
@@ -763,4 +764,51 @@ describe('PerformanceMonitoring', function() {
...
+      expect(payload[0].spans[0].context.destination).toBeDefined()
+      expect(payload[0].spans[1].context.destination).toBeDefined()
+
+      apmServer
+        .sendTransactions(payload)
+        .catch(reason => {
+          fail(
```
Comment (jahtalab) [inline packages/rum-core/test/performance-monitoring/performance-monitoring.spec.js]: Please use `done.fail()` instead, as discussed in previous PRs, this won't end test properly.

Comment (vigneshshanmugam) [reply packages/rum-core/test/performance-monitoring/performance-monitoring.spec.js]: Wow didn't know something like that exists, for sure 👍

**Comment by jahtalab** [comment]
@vigneshshanmugam this doesn't close elastic/apm-agent-rum-js#513, that's a separate feature.

**Pull request merged**